domain/network/state/subnet.go

```diff
@@ -20,31 +20,53 @@ const (
 	subnetTypeFanOverlaySegment = 1
 )
 
-// AddSubnet creates a subnet.
-func (st *State) AddSubnet(
-	ctx context.Context,
-	uuid utils.UUID,
-	cidr string,
-	providerID network.Id,
-	providerNetworkID network.Id,
-	VLANTag int,
-	availabilityZones []string,
-	spaceUUID string,
-	fanInfo *network.FanCIDRs,
-) error {
+// UpsertSubnets updates or adds each one of the provided subnets in one
+// transaction.
+func (st *State) UpsertSubnets(ctx context.Context, subnets []network.SubnetInfo) error {
 	db, err := st.DB()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
+	return db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		for _, subnet := range subnets {
+			err := updateSubnetSpaceID(
+				ctx,
+				tx,
+				string(subnet.ID),
+				subnet.SpaceID,
+			)
+			if err != nil && !errors.Is(err, errors.NotFound) {
+				return errors.Trace(err)
+			}
+			// If the subnet doesn't exist yet we need to create it.
+			if errors.Is(err, errors.NotFound) {
+				if err := addSubnet(
+					ctx,
+					tx,
+					subnet,
+				); err != nil {
+					return errors.Trace(err)
+				}
+			}
+		}
+		return nil
+	})
+}
+
+func addSubnet(
+	ctx context.Context,
+	tx *sql.Tx,
+	subnet network.SubnetInfo,
+) error {
 	var subnetType int
-	if fanInfo != nil {
+	if subnet.FanInfo != nil {
 		subnetType = subnetTypeFanOverlaySegment
 	}
 	var spaceUUIDValue any
-	spaceUUIDValue = spaceUUID
-	if spaceUUID == "" {
-		spaceUUIDValue = nil
+	spaceUUIDValue = subnet.SpaceID
+	if subnet.SpaceID == "" {
+		spaceUUIDValue = network.AlphaSpaceId
```

Comment:
You can remove the following line `var spaceUUIDValue any` and just do the following:

```suggestion
	spaceUUIDValue := subnet.SpaceID
	if subnet.SpaceID == "" {
		spaceUUIDValue = network.AlphaSpaceId
```